manifests/init.pp

```diff
@@ -214,7 +218,7 @@
     }
   }
 
-  if $pulp_master or $pulp {
+  if ($pulp_master or $pulp) and $facts['os']['release']['major'] == '7' {
```

Comment:
All these checks are now redundant since we fail early on.

Reply:
Safety first!

Reply:
That is me saying, the fail does cover it but I also see this has helping to cover our bases and make it clear in the code this block is EL7 only. I debated moving a bunch of htis to their own class to highlight that as well but didn't want to do too much re-factoring.

Reply:
Feels a bit redundant to me since you know there's no Pulp 2 on EL8 anyway.

Reply:
I know that, but doesn't mean everyone reading this code will clearly know that.

Reply:
Partly I agree, but the code is already very hard to follow. Adding more conditionals just makes it harder rather than easier IMHO. I'm waiting till we can drop Pulp for a big refactor.